graphql/src/main/java/com/scalar/db/graphql/datafetcher/DataFetcherBase.java

```diff
@@ -0,0 +1,207 @@
+    DistributedTransaction transaction = getTransactionIfEnabled(environment);
+    if (transaction != null) {
+      return transaction.scan(scan.withConsistency(Consistency.LINEARIZABLE));
+    } else {
+      return ImmutableList.copyOf(storage.scan(scan));
+    }
+  }
+
+  protected void performPut(DataFetchingEnvironment environment, Put put)
+      throws TransactionException, ExecutionException {
+    DistributedTransaction transaction = getTransactionIfEnabled(environment);
+    if (transaction != null) {
+      transaction.put(put);
+    } else {
+      storage.put(put);
+    }
+  }
+
+  protected void performDelete(DataFetchingEnvironment environment, Delete delete)
+      throws TransactionException, ExecutionException {
+    DistributedTransaction transaction = getTransactionIfEnabled(environment);
+    if (transaction != null) {
+      transaction.delete(delete.withConsistency(Consistency.LINEARIZABLE));
+    } else {
+      storage.delete(delete);
+    }
+  }
+
+  private DistributedTransaction getTransactionIfEnabled(DataFetchingEnvironment environment) {
+    DistributedTransaction transaction = null;
+    if (tableModel.getTransactionEnabled()) {
+      transaction = environment.getGraphQlContext().get(Constants.CONTEXT_TRANSACTION_KEY);
+    }
+    return transaction;
+  }
+
+  /**
+   * Create a list of Scalar DB {@link ConditionalExpression} objects from a list of
+   * ConditionalExpression in GraphQL request.
+   *
+   * @param graphQlConditionalExpressions expressions field input in PutCondition or DeleteCondition
+   * @return A list of {@link ConditionalExpression}
+   */
+  protected List<ConditionalExpression> getConditionalExpressions(
+      List<Map<String, Object>> graphQlConditionalExpressions) {
+    return graphQlConditionalExpressions.stream()
+        .map(
+            ex ->
+                new ConditionalExpression(
+                    (String) ex.get("name"),
+                    createAnonymousValueFromMap(ex),
+                    ConditionalExpression.Operator.valueOf((String) ex.get("operator"))))
+        .collect(toList());
+  }
+
+  protected Value<?> createAnonymousValueFromMap(Map<String, Object> map) {
+    Object v = getOneScalarValue(map);
+    Value<?> value;
+    if (v instanceof Integer) {
+      value = new IntValue((Integer) v);
+    } else if (v instanceof Long) {
+      value = new BigIntValue((Long) v);
+    } else if (v instanceof Float) {
+      value = new FloatValue((Float) v);
+    } else if (v instanceof Double) {
+      value = new DoubleValue((Double) v);
+    } else if (v instanceof String) {
+      value = new TextValue((String) v);
+    } else if (v instanceof Boolean) {
+      value = new BooleanValue((Boolean) v);
+    } else {
+      throw new IllegalArgumentException("Unexpected value: " + v.getClass());
+    }
+    return value;
+  }
+
+  protected Value<?> createNamedValueFromMap(String name, Map<String, Object> map) {
```

Comment:
I think we can unify this and the above method into one to avoid redundancy.
How about giving "" to XxxxValue constructor when name is null?

Reply:
Thanks! Unified two methods in 05f9332.